pydatalab/src/pydatalab/apps/uvvis/__init__.py

```diff
@@ -0,0 +1,170 @@
+import os
+from pathlib import Path
+
+import bokeh.embed
+import numpy as np
+import pandas as pd
+from bokeh.models import HoverTool
+
+from pydatalab.blocks.base import DataBlock
+from pydatalab.bokeh_plots import DATALAB_BOKEH_THEME, selectable_axes_plot
+from pydatalab.file_utils import get_file_info_by_id
+from pydatalab.logger import LOGGER
+
+
+class UVVisBlock(DataBlock):
+    accepted_file_extensions = (".txt", ".TXT")
+    blocktype = "uvvis"
+    name = "UV-Vis"
+    description = (
+        "This block can plot UV-Vis data from a .txt file. "
+        "Two files are required, the scan to plot and the background scan.\n"
+        "The first file in the order will be treated as the background scan, and subsequent files as the sample scans."
+    )
+
+    @property
+    def plot_functions(self):
+        return (self.generate_absorbance_plot,)
+
+    @classmethod
+    def parse_uvvis_txt(self, filename: Path) -> pd.DataFrame:
+        """
+        Parses a UV-Vis .txt file into a pandas DataFrame
+        Args:
+            filename (Path): Path to the .txt file
+        Returns:
+            pd.DataFrame: DataFrame containing the UV-Vis data with columns for wavelength and absorbance
+        """
+        # Read the file, skipping the first 7 rows and using the first row as header
+        data = pd.read_csv(filename, sep=r";", skiprows=7, header=None)
```

Comment:
Definitely always 7? Alternatively we could look for the blank lines around the header that these files all seem to have